rest-api/src/main/kotlin/fund/cyber/markets/api/rest/service/PriceService.kt

```diff
@@ -0,0 +1,74 @@
+package fund.cyber.markets.api.rest.service
+
+import fund.cyber.markets.cassandra.common.toTokenTicker
+import fund.cyber.markets.cassandra.repository.TickerRepository
+import fund.cyber.markets.common.Intervals
+import fund.cyber.markets.common.MILLIS_TO_DAYS
+import fund.cyber.markets.common.closestSmallerMultiplyFromTs
+import fund.cyber.markets.common.convert
+import fund.cyber.markets.ticker.common.CrossConversion
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.core.publisher.toFlux
+import reactor.core.publisher.toMono
+import java.math.BigDecimal
+import java.util.*
+
+@Service
+class PriceService(
+    private val tickerRepository: TickerRepository
+) {
+
+    fun getPrices(base: String, quoutes: List<String>, exchange: String): Mono<Map<String, BigDecimal>> {
+
+        val timestampFrom = Date(closestSmallerMultiplyFromTs(Intervals.MINUTE) - Intervals.MINUTE)
+        val timestampTo = Date(closestSmallerMultiplyFromTs(Intervals.MINUTE))
+        val epochDay = timestampFrom.time convert MILLIS_TO_DAYS
+
+        var tickersFlux = tickerRepository
+            .find(base, epochDay, timestampFrom, timestampTo, Intervals.MINUTE)
+
+        quoutes.forEach { quote ->
```

Comment:
quoutes - naming mistake 
Use mergeWith( quotes.toFlux().flatMap{...} ) on first Flux instead of for each

Reply:
done